lib/ops/events/codes.go

```diff
@@ -0,0 +1,430 @@
+		Code: TLSKeyPairCreatedCode,
+	}
+	// TLSKeyPairDeleted is emitted when cluster web certificate is deleted.
+	TLSKeyPairDeleted = events.Event{
+		Name: TLSKeyPairDeletedEvent,
+		Code: TLSKeyPairDeletedCode,
+	}
+	// AuthPreferenceUpdated is emitted when cluster auth preference is updated.
+	AuthPreferenceUpdated = events.Event{
+		Name: AuthPreferenceUpdatedEvent,
+		Code: AuthPreferenceUpdatedCode,
+	}
+	// SMTPConfigCreated is emitted when SMTP configuration is created/updated.
+	SMTPConfigCreated = events.Event{
+		Name: SMTPConfigCreatedEvent,
+		Code: SMTPConfigCreatedCode,
+	}
+	// SMTPConfigDeleted is emitted when SMTP configuration is deleted.
+	SMTPConfigDeleted = events.Event{
+		Name: SMTPConfigDeletedEvent,
+		Code: SMTPConfigDeletedCode,
+	}
+	// AlertCreated is emitted when monitoring alert is created/updated.
+	AlertCreated = events.Event{
+		Name: AlertCreatedEvent,
+		Code: AlertCreatedCode,
+	}
+	// AlertDeleted is emitted when monitoring alert is deleted.
+	AlertDeleted = events.Event{
+		Name: AlertDeletedEvent,
+		Code: AlertDeletedCode,
+	}
+	// AlertTargetCreated is emitted when monitoring alert target is created/updated.
+	AlertTargetCreated = events.Event{
+		Name: AlertTargetCreatedEvent,
+		Code: AlertTargetCreatedCode,
+	}
+	// AlertTargetDeleted is emitted when monitoring alert target is deleted.
+	AlertTargetDeleted = events.Event{
+		Name: AlertTargetDeletedEvent,
+		Code: AlertTargetDeletedCode,
+	}
+	// AuthGatewayUpdated is emitted when cluster auth gateway settings are updated.
+	AuthGatewayUpdated = events.Event{
+		Name: AuthGatewayUpdatedEvent,
+		Code: AuthGatewayUpdatedCode,
+	}
+	// UserInviteCreated is emitted when a user invite is created.
+	UserInviteCreated = events.Event{
+		Name: InviteCreatedEvent,
+		Code: UserInviteCreatedCode,
+	}
+	// ClusterUnhealthy is emitted when cluster becomes unhealthy.
+	ClusterUnhealthy = events.Event{
+		Name: ClusterDegradedEvent,
+		Code: ClusterUnhealthyCode,
+	}
+	// ClusterHealthy is emitted when cluster becomes healthy.
+	ClusterHealthy = events.Event{
+		Name: ClusterActivatedEvent,
+		Code: ClusterHealthyCode,
+	}
+	// ApplicationInstall is emitted when a new application image is installed.
+	ApplicationInstall = events.Event{
+		Name: AppInstalledEvent,
+		Code: ApplicationInstallCode,
+	}
+	// ApplicationUpgrade is emitted when an application release is upgraded.
+	ApplicationUpgrade = events.Event{
+		Name: AppUpgradedEvent,
+		Code: ApplicationUpgradeCode,
+	}
+	// ApplicationRollback is emitted when an application release is rolled back.
+	ApplicationRollback = events.Event{
+		Name: AppRolledBackEvent,
+		Code: ApplicationRollbackCode,
+	}
+	// ApplicationUninstall is emitted when an application release is uninstalled.
+	ApplicationUninstall = events.Event{
+		Name: AppUninstalledEvent,
+		Code: ApplicationUninstallCode,
+	}
+)
+
+const (
+	// OpereationInstallStartCode is the install operation start event code.
+	OperationInstallStartCode = "G0001I"
```

Comment:
Should we also remove the severity suffixes from the code?

Reply:
I think it's fine to keep it in the code. Makes it convenient to use the same number code for different outcomes of the same thing (e.g. operation succeeded/failed, etc.).